### PR TITLE
Sohaib models 1

### PR DESCRIPTION
**Pull request by SohaibkhanFolio3**
This pull request adds three new SQLAlchemy models to the project: `Books`, `Genre`, and `Book_Genre`. These models are used to represent books and their associated genres in the database.

`Books` includes attributes for the book's ID, title, date of publication, ISBN, description, language ID, creation time, and update time. `Genre` includes attributes for the genre's ID, name, creation time, and update time. `Book_Genre` includes attributes for the association between a book and a genre, as well as the creation time.

In addition to the model code, this pull request also adds pre-commit hooks to automatically format the code using `flake8`, `black`, and `isort` before committing. This helps ensure that the code adheres to a consistent style and catches potential errors before they are committed.

Tickets: 
https://app.asana.com/0/1204411152293714/1204422166915905/f
https://app.asana.com/0/1204411152293714/1204422166915917/f
https://app.asana.com/0/1204411152293714/1204422166915919/f

### CODE REVIEW

**Review by mtalhatahir2001 — APPROVED** [review]

File: src/models/book.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        description (str): A description of the book.
+        language_id (int): The ID of the language the book is written in.
+        created_at (datetime): The date and time the book was created.
+        updated_at (datetime): The date and time the book was last updated.
+    """
+
+    __tablename__ = "books"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    title: Mapped[str] = mapped_column(String(32), unique=True, nullable=False)
```
Comment (mustansirMuz) [inline src/models/book.py:26]: Isn't 32 a very short length for title? I think titles can be lengthier than that...

Also we don't necessarily need to provide lengths imo

File: src/models/book.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    date_of_publication: Mapped[datetime] = mapped_column(
+        DateTime(), nullable=False
+    )  # noqa
+    isbn: Mapped[str] = mapped_column(Integer)
```
Comment (mustansirMuz) [inline src/models/book.py:30]: ISBN should be a string as it can have hyphens as well, also it should be unique...

**Review by mustansirMuz — CHANGES_REQUESTED** [review]

File: src/models/book.py (hunk 1 above)
Comment (SohaibkhanFolio3) [reply src/models/book.py:26]: Talha suggested 32 for strings ! I ll fix this .

File: src/models/book.py (hunk 2 above)
Comment (SohaibkhanFolio3) [reply src/models/book.py]: Resolved

**Review by mustansirMuz — APPROVED** [review]
LGTM

**Pull request merged**